# Grace Crandall 
## FISH 497 - Spring 2019

#### Notes/data: 
Two blast outputs, taxids on last column

http://gannet.fish.washington.edu/Atumefaciens/20190416_metagenomics_pgen_blastn/pH71.blastn.outfmt6

http://gannet.fish.washington.edu/Atumefaciens/20190416_metagenomics_pgen_blastn/pH82.blastn.outfmt6

---

This repository contains the data, scripts, and analyses related to identifying the proteins related to non-oyster cells in order to create a table of potential non-oyster organisms that can be found in the water that the oytsers are reared in.

## Repository contents:
---
### [data](https://github.com/fish497-2019/GraceCrandall/tree/master/data)       
- Contains deduced proteomes from .... 

### [scripts](https://github.com/fish497-2019/GraceCrandall/tree/master/scripts)      
- Contains R scripts from coursework and project analyses

### [analyses](https://github.com/fish497-2019/GraceCrandall/blob/master/analyses/readme.md)     
- Contains tables and figures from analyses related to my FISH 497 project


---
## Plan for analyses:
---
1. Use the deduced proteome as the background for my Oyster 2015 dataset in order to identify data related to the organisms that shared the water with the oysters ([GitHub Issue: #693](https://github.com/RobertsLab/resources/issues/693))    
2. Use the resulting dataframe to ID the taxa that the proteins belong to using BLAST (taxonomy). 
3. Create a table that identifies the taxa found.
4. Create other plots and visualizations related to the taxa/proteins/etc. 
