06042019-project
================

R Markdown Script for Working with taxonomic microbiome data
------------------------------------------------------------

This project will work towards identifying the organisms living in the water at hatcheries where oysters and other shellfish are reared.

Required packages
-----------------

Load in data from <https://gannet.fish.washington.edu/Atumefaciens/20190516_metagenomics_pgen_blastx/>
------------------------------------------------------------------------------------------------------

### Use files with `outfmt6`

``` r
pH71 <- read.csv("https://gannet.fish.washington.edu/Atumefaciens/20190516_metagenomics_pgen_blastx/MG3_pH71.blastx.outfmt6", sep = '\t', header = FALSE)
pH82 <- read.csv("https://gannet.fish.washington.edu/Atumefaciens/20190516_metagenomics_pgen_blastx/MG1_pH82.blastx.outfmt6", sep = '\t', header = FALSE)
```

Add Column names (it's a blast output, last column in taxonomy genus\_species info)
-----------------------------------------------------------------------------------

``` r
colnames(pH71)
```

    ##  [1] "V1"  "V2"  "V3"  "V4"  "V5"  "V6"  "V7"  "V8"  "V9"  "V10" "V11"
    ## [12] "V12" "V13" "V14"

``` r
cols <- c("V1", "V2", "V3", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "tax_ID")
colnames(pH71) <- cols
```

Version from Sam (<https://github.com/RobertsLab/resources/issues/698>)
-----------------------------------------------------------------------

``` r
taxa_counts <- pH71 %>% 
  group_by(tax_ID) %>% 
  summarise(count = n())
taxa_counts
```

    ## # A tibble: 2,929 x 2
    ##    tax_ID                                                count
    ##    <fct>                                                 <int>
    ##  1 [Bacillus thuringiensis] serovar konkukian str. 97-27     5
    ##  2 [Bacillus] caldolyticus                                   3
    ##  3 [Bacillus] caldotenax                                    15
    ##  4 [Bacillus] caldovelox                                     4
    ##  5 [Candida] glabrata CBS 138                               39
    ##  6 [Clostridium] bolteae ATCC BAA-613                        1
    ##  7 [Clostridium] saccharolyticum WM1                         3
    ##  8 [Clostridium] scindens                                   10
    ##  9 [Clostridium] scindens ATCC 35704                         2
    ## 10 [Clostridium] symbiosum                                  30
    ## # … with 2,919 more rows

Make plots showing abundances... cut down to those that show up more than 1000 times
------------------------------------------------------------------------------------

``` r
smaller <- taxa_counts %>%
  filter(taxa_counts$count > 1000)
smaller
```

    ## # A tibble: 29 x 2
    ##    tax_ID                                     count
    ##    <fct>                                      <int>
    ##  1 Acanthamoeba polyphaga mimivirus            1002
    ##  2 Arabidopsis thaliana                        3796
    ##  3 Bacillus subtilis subsp. subtilis str. 168  7302
    ##  4 Bacteroides thetaiotaomicron VPI-5482       1126
    ##  5 Bos taurus                                  1789
    ##  6 Caenorhabditis elegans                      1360
    ##  7 Candidatus Pelagibacter ubique HTCC1062     3852
    ##  8 Cytophaga hutchinsonii ATCC 33406           1315
    ##  9 Danio rerio                                 1540
    ## 10 Dictyostelium discoideum                    2009
    ## # … with 19 more rows

### create bar plot of 29 most abundant taxa and their counts (pH7.1)

``` r
ggplot(data=smaller, aes(x=tax_ID, y=count)) +
  geom_bar(stat="identity")
```

![](060419-mydata_files/figure-markdown_github/pH7.1%20taxa%20counts%201000%20times%20or%20more-1.png)

Do the same for pH 8.2
----------------------

### Rename last column tax\_ID

``` r
colnames(pH82)
```

    ##  [1] "V1"  "V2"  "V3"  "V4"  "V5"  "V6"  "V7"  "V8"  "V9"  "V10" "V11"
    ## [12] "V12" "V13" "V14"

``` r
cols <- c("V1", "V2", "V3", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "tax_ID")
colnames(pH82) <- cols
```

get the counts for the unique taxa IDs
--------------------------------------

``` r
taxa_counts82 <- pH82 %>% 
  group_by(tax_ID) %>% 
  summarise(count = n())
taxa_counts82
```

    ## # A tibble: 2,943 x 2
    ##    tax_ID                                                count
    ##    <fct>                                                 <int>
    ##  1 [Bacillus thuringiensis] serovar konkukian str. 97-27    19
    ##  2 [Bacillus] caldolyticus                                   6
    ##  3 [Bacillus] caldotenax                                    10
    ##  4 [Bacillus] caldovelox                                     2
    ##  5 [Candida] boidinii                                        2
    ##  6 [Candida] glabrata CBS 138                               68
    ##  7 [Candida] oleophila                                       1
    ##  8 [Clostridium] saccharolyticum WM1                         3
    ##  9 [Clostridium] scindens                                   30
    ## 10 [Clostridium] scindens ATCC 35704                         1
    ## # … with 2,933 more rows

create smaller table of those that show up more than 1000 times
---------------------------------------------------------------

``` r
smaller82 <- taxa_counts82 %>%
  filter(taxa_counts82$count > 1000)
smaller82
```

    ## # A tibble: 50 x 2
    ##    tax_ID                                     count
    ##    <fct>                                      <int>
    ##  1 Acanthamoeba polyphaga mimivirus            1972
    ##  2 Agrobacterium fabrum str. C58               1633
    ##  3 Alcanivorax borkumensis SK2                 1131
    ##  4 Aquifex aeolicus VF5                        1802
    ##  5 Arabidopsis thaliana                        8246
    ##  6 Azospirillum brasilense                     1082
    ##  7 Bacillus subtilis subsp. subtilis str. 168 13247
    ##  8 Bacteroides thetaiotaomicron VPI-5482       1247
    ##  9 Bos taurus                                  1400
    ## 10 Bradyrhizobium diazoefficiens USDA 110      1676
    ## # … with 40 more rows

plot number of times the top 50 show up in pH8.2
------------------------------------------------

``` r
ggplot(data=smaller82, aes(x=tax_ID, y=count)) +
  geom_bar(stat="identity")
```

![](060419-mydata_files/figure-markdown_github/pH8.2%20plot%20of%20taxa%20that%20occur%201000%20or%20more%20times-1.png)

Compare pH7.1 and pH8.2
-----------------------

### make a table showing the organisms that are present in both pH7.1 and pH8.2 (only from the lists that contain those with 1000+ occurrences)

``` r
shared <- inner_join(smaller, smaller82, by = "tax_ID")
```

    ## Warning: Column `tax_ID` joining factors with different levels, coercing to
    ## character vector

``` r
shared
```

    ## # A tibble: 26 x 3
    ##    tax_ID                                     count.x count.y
    ##    <chr>                                        <int>   <int>
    ##  1 Acanthamoeba polyphaga mimivirus              1002    1972
    ##  2 Arabidopsis thaliana                          3796    8246
    ##  3 Bacillus subtilis subsp. subtilis str. 168    7302   13247
    ##  4 Bacteroides thetaiotaomicron VPI-5482         1126    1247
    ##  5 Bos taurus                                    1789    1400
    ##  6 Caenorhabditis elegans                        1360    1082
    ##  7 Candidatus Pelagibacter ubique HTCC1062       3852   19601
    ##  8 Cytophaga hutchinsonii ATCC 33406             1315    1059
    ##  9 Dictyostelium discoideum                      2009    3185
    ## 10 Drosophila melanogaster                       2638    1159
    ## # … with 16 more rows

### rename columns

``` r
cols2 <- c("tax_ID", "pH71", "pH82")
colnames(shared) <- cols2
```

### gather the shared data so that the counts are all in one column, and the treatments are in another. count.x --&gt; pH7.1, count.y --&gt; pH8.2

``` r
shared2 <- shared %>% 
  gather(`pH71`, `pH82`, key = "pH", value = "counts")
shared2
```

    ## # A tibble: 52 x 3
    ##    tax_ID                                     pH    counts
    ##    <chr>                                      <chr>  <int>
    ##  1 Acanthamoeba polyphaga mimivirus           pH71    1002
    ##  2 Arabidopsis thaliana                       pH71    3796
    ##  3 Bacillus subtilis subsp. subtilis str. 168 pH71    7302
    ##  4 Bacteroides thetaiotaomicron VPI-5482      pH71    1126
    ##  5 Bos taurus                                 pH71    1789
    ##  6 Caenorhabditis elegans                     pH71    1360
    ##  7 Candidatus Pelagibacter ubique HTCC1062    pH71    3852
    ##  8 Cytophaga hutchinsonii ATCC 33406          pH71    1315
    ##  9 Dictyostelium discoideum                   pH71    2009
    ## 10 Drosophila melanogaster                    pH71    2638
    ## # … with 42 more rows

create bar plots of shared organsims (26 shared organisms)
----------------------------------------------------------

``` r
ggplot(shared2, aes(fill = pH, y=counts, x=tax_ID)) + 
    geom_bar(position="dodge", stat="identity")
```

![](060419-mydata_files/figure-markdown_github/shared%20highly%20abundance%20organisms%20in%20pH7.1%20and%20pH8.2-1.png)

add column in shared of absolute values of differences between the abundances of the organisms
----------------------------------------------------------------------------------------------

``` r
shared3 <- mutate(shared, 
  abs_diff = abs(shared$pH71 - shared$pH82))
shared3
```

    ## # A tibble: 26 x 4
    ##    tax_ID                                      pH71  pH82 abs_diff
    ##    <chr>                                      <int> <int>    <int>
    ##  1 Acanthamoeba polyphaga mimivirus            1002  1972      970
    ##  2 Arabidopsis thaliana                        3796  8246     4450
    ##  3 Bacillus subtilis subsp. subtilis str. 168  7302 13247     5945
    ##  4 Bacteroides thetaiotaomicron VPI-5482       1126  1247      121
    ##  5 Bos taurus                                  1789  1400      389
    ##  6 Caenorhabditis elegans                      1360  1082      278
    ##  7 Candidatus Pelagibacter ubique HTCC1062     3852 19601    15749
    ##  8 Cytophaga hutchinsonii ATCC 33406           1315  1059      256
    ##  9 Dictyostelium discoideum                    2009  3185     1176
    ## 10 Drosophila melanogaster                     2638  1159     1479
    ## # … with 16 more rows

create bar plot of absolute differences
---------------------------------------

``` r
ggplot(data=shared3, aes(x=tax_ID, y=abs_diff)) +
  geom_bar(stat="identity")
```

![](060419-mydata_files/figure-markdown_github/absolute%20differences%20between%20highly%20abundant%20organism%20counts%20in%20pH%207.1%20and%20pH%208.2-1.png)

sort shared3 (shared highly abundant organisms) by absolute difference
----------------------------------------------------------------------

``` r
shared4 <- arrange(shared3, abs_diff, tax_ID, pH71, pH82)

#filter the top two taxa by having a cut-off of 6000 difference
filter(shared4, abs_diff > 6000)
```

    ## # A tibble: 2 x 4
    ##   tax_ID                                   pH71  pH82 abs_diff
    ##   <chr>                                   <int> <int>    <int>
    ## 1 Escherichia coli K-12                    8632 19149    10517
    ## 2 Candidatus Pelagibacter ubique HTCC1062  3852 19601    15749

tables of unique organisms in pH 7.1 and pH 8.2
-----------------------------------------------

``` r
uniq71 <- left_join(smaller, smaller82, by = "tax_ID")
```

    ## Warning: Column `tax_ID` joining factors with different levels, coercing to
    ## character vector

``` r
#3 unique in 7.1 
filter(uniq71, count.y == "NA")
```

    ## # A tibble: 0 x 3
    ## # … with 3 variables: tax_ID <chr>, count.x <int>, count.y <int>

``` r
uniq82 <- left_join(smaller82, smaller, by = "tax_ID")
```

    ## Warning: Column `tax_ID` joining factors with different levels, coercing to
    ## character vector

``` r
#24 unique in 8.2
filter(uniq82, count.y == 'NA')
```

    ## # A tibble: 0 x 3
    ## # … with 3 variables: tax_ID <chr>, count.x <int>, count.y <int>
