---
title: "06042019-project"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown Script for Working with taxonomic microbiome data
This project will works towards identifying the organisms living in the water at hatcheries where oysters and other shellfish are reared. 

## Required packages
```{r}
library(dplyr)
library(ggplot2)
library(tidyr)
```

## Load in data from https://gannet.fish.washington.edu/Atumefaciens/20190516_metagenomics_pgen_blastx/ 
### Use files with `outfmt6` 
```{r}
pH71 <- read.csv("https://gannet.fish.washington.edu/Atumefaciens/20190516_metagenomics_pgen_blastx/MG3_pH71.blastx.outfmt6", sep = '\t', header = FALSE)
pH82 <- read.csv("https://gannet.fish.washington.edu/Atumefaciens/20190516_metagenomics_pgen_blastx/MG1_pH82.blastx.outfmt6", sep = '\t', header = FALSE)
```
## Add Column names (it's a blast output)
```{r}


```
## Figure out a way to get species be columns with row value as count
### create issues to ask questions!

## other options: evalue --> use spread to get the average evalues for each species, etc. focues on previous idea first though

