---
title: "06042019-project"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown Script for Working with taxonomic microbiome data
This project will work towards identifying the organisms living in the water at hatcheries where oysters and other shellfish are reared. 

## Required packages
```{r}
library(dplyr)
library(ggplot2)
library(tidyr)
```

## Load in data from https://gannet.fish.washington.edu/Atumefaciens/20190516_metagenomics_pgen_blastx/ 
### Use files with `outfmt6` 
```{r}
pH71 <- read.csv("https://gannet.fish.washington.edu/Atumefaciens/20190516_metagenomics_pgen_blastx/MG3_pH71.blastx.outfmt6", sep = '\t', header = FALSE)
pH82 <- read.csv("https://gannet.fish.washington.edu/Atumefaciens/20190516_metagenomics_pgen_blastx/MG1_pH82.blastx.outfmt6", sep = '\t', header = FALSE)
```
## Add Column names (it's a blast output, last column in taxonomy genus_species info)
## Option 1
```{r}
pH71 %>%
  group_by(V14) %>%
  tally()
```

## Version from Sam (https://github.com/RobertsLab/resources/issues/698)
```{r}
taxa_counts <- pH71 %>% 
  group_by(pH71$V14) %>% 
  summarise(count = n())
```

## Make plots showing abundances... maybe cut down list to those that have 2 digits or more in counts
```{r}
ggplot(data = pH71) + 
  geom_bar(mapping = aes(x = V14, color = V14))
```


